internal/compiler-interface/src/main/java/xsbti/compile/CachedCompilerProvider.java

```diff
@@ -22,5 +22,5 @@
 	ScalaInstance scalaInstance();
 
 	/** Return a new cached compiler from the usual compiler input. */
-	CachedCompiler newCachedCompiler(String[] arguments, Output output, Logger log, Reporter reporter);
+	Object newCachedCompiler(String[] arguments, Output output, Logger log, Reporter reporter);
```

Comment:
Maybe return `Optional<CachedCompiler>` and add `newCachedCompiler2` that returns `Optional<CachedCompiler2>`?

Reply:
That still leaves the possibility of both returning empty - then what?

Reply:
Throw?

Reply:
Either way, we can work on that in a follow-up PR to remove cached compiler in general.